File: backend/src/main/java/ch/rodano/core/services/bll/configurator/ReportGrantsService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.Map;
import java.util.UUID;

public interface ReportGrantsService {

	Map<UUID, List<UUID>> getReportGrants(UUID projectId);

	void saveReportGrants(UUID projectId, Map<UUID, List<UUID>> profileReportMap);
}

File: frontends/main/src/app/configurator/services/api/report-grants.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({providedIn: 'root'})
export class ReportGrantsService {
	constructor(private http: HttpClient) {}

	getReportGrants(projectId: string): Observable<Record<string, string[]>> {
		return this.http.get<Record<string, string[]>>(`/api/superuser/configurator/projects/${projectId}/config/report-grants`);
	}

	saveReportGrants(projectId: string, grants: Record<string, string[]>): Observable<void> {
		return this.http.put<void>(`/api/superuser/configurator/projects/${projectId}/config/report-grants`, grants);
	}
}
